File: base/src/main/java/com/example/base/BaseActivity.java
package com.example.base;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.base.view.IBaseView;
import com.example.commen.ActivityInstanceManager;
import com.example.commen.NetConnectManager;
import com.example.commen.P2PError;

import java.util.List;

public abstract class BaseActivity<T> extends AppCompatActivity implements NetConnectManager.INetConnectListener, IBaseView<T> {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());

        initView();

        initData(); //一般在子线程中操作

        ActivityInstanceManager.addActivity(this);

        NetConnectManager.getInstance().registerNetConnectListener(this);
    }

    public boolean isConnected(){
        return NetConnectManager.getInstance().isConnectStatus();
    }

    //让子类提供布局
    protected abstract int getLayoutId();

    //初始化view
    protected abstract void initView();

    //初始化数据
    protected abstract void initData();


    /**
     * 网路状态
     */
    @Override
    public void onConnected() {

    }

    @Override
    public void onDisConnected() {

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        ActivityInstanceManager.removeActivity(this);
        NetConnectManager.getInstance().unRegisterNetConnectListener(this);
    }

    @Override
    public void onHttpRequestDataSuccess(int requestCode, T data) {

    }

    @Override
    public void onHttpRequestDataListSuccess(int requestCode, List<T> data) {

    }

    @Override
    public void onHttpRequestDataFailed(int requestCode, P2PError error) {

    }

    @Override
    public void showLoading(int showCode) {

    }

    @Override
    public void hideLoading(int showCode) {

    }
}

File: moduleBase/src/main/java/com/example/modulebase/BaseActivity.kt
package com.example.modulebase

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.modulecommon.Constructor
import com.example.modulecommon.manager.AppManager

import com.example.modulecommon.manager.NetConnetMannager
import com.gyf.immersionbar.ImmersionBar
import com.gyf.immersionbar.ktx.immersionBar


//Activity基类
abstract class BaseActivity : AppCompatActivity(), NetConnetMannager.INetConnectListener {
    override fun onConnected() {

    }

    override fun onDisConnected() {

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        flagFullScreen()
        setContentView(getLayoutId())
        ImmersionBar.with(this).init()
        initTitle()
        initData()
        initTab()
        NetConnetMannager.getInstance().registerNetConnectListener(this)//注册监听网络连接
        AppManager.getInstance().add(this)
    }

    open fun flagFullScreen(){}

    open fun isConnected():Boolean{
       return NetConnetMannager.getInstance().isConnectStatus
    }
    open fun initTab(){}

     open fun initData(){}

     open fun initTitle(){}

    abstract fun  getLayoutId(): Int

    override fun onDestroy() {
        super.onDestroy()
        AppManager.getInstance().remove(this)
        NetConnetMannager.getInstance().unRegisterNetConnerctListener(this)
    }
    //读取用户信息
    open fun readUser() : User{
        var sp = this.getSharedPreferences(Constructor.SP_NAME_USERINFO, Context.MODE_PRIVATE)
        var user = User()
        user.name = sp.getString("name", "")
        user.imageurl = sp.getString("imageurl", "")
        user.phone = sp.getString("phone", "")
        user.isCredit = sp.getBoolean("iscredit", false)

        return user
    }

    //启动新的activity
    fun goToActivity(Activity: Class<*>, bundle: Bundle?) {
        val intent = Intent(this, Activity)
        //携带数据
        if (bundle != null && bundle.size() != 0) {
            intent.putExtra(Constructor.BUNDLE, bundle)
        }

        startActivity(intent)
    }
    //保存用户信息
  open  fun saveUser(user: User) {
        var sp = this.getSharedPreferences(Constructor.SP_NAME_USERINFO, Context.MODE_PRIVATE)
        var editor = sp.edit()
        editor.putString("name", user.name)
        editor.putString("imageurl", user.imageurl)
        editor.putBoolean("iscredit", user.isCredit)
        editor.putString("phone", user.phone)
        editor.apply()//必须提交，否则保存不成功
    }
}
